plugin.video.showcase.canada/lib/scraper.py

```diff
@@ -0,0 +1,158 @@
+#
+from t1mlib import t1mAddon
+import json
+import re
+import os
+import datetime
+import urllib
+import urllib2
+import xbmc
+import xbmcplugin
+import xbmcgui
+import HTMLParser
+import sys
+
+h = HTMLParser.HTMLParser()
+qp  = urllib.quote_plus
+uqp = urllib.unquote_plus
+UTF8     = 'utf-8'
+
+
+class myAddon(t1mAddon):
+
+  def getAddonMenu(self,url,ilist):
+   addonLanguage  = self.addon.getLocalizedString
+   url = 'http://common.farm1.smdg.ca/Forms/PlatformVideoFeed?platformUrl=http%3A//feed.theplatform.com/f/dtjsEC/9H6qyshBZU3E/categories%3Fpretty%3Dtrue%26byHasReleases%3Dtrue%26byCustomValue%3D%7Bplayertag%7D%7Bz/Showcase%20Video%20Centre%7D%26sort%3DfullTitle&callback='
+   html = self.getRequest(url)
+   a = json.loads(html[1:len(html)-1])['items']
+   wewait = True
+   for b in a:
+    if b['title'] == 'Shows' and wewait == True:
+        wewait = False
+        continue
+    if wewait == False:
+      if b['depth'] == 2:
+       name = b['title']
+       url  = b['id'].rsplit('/',1)[1]
+       ilist = self.addMenuItem(name,'GC', ilist, url, self.addonIcon, self.addonFanart, {}, isFolder=True)
+   return(ilist)
+
+  def getAddonCats(self,url,ilist):
+   self.defaultVidStream['width']  = 640
+   self.defaultVidStream['height'] = 360
+   geurl = uqp(url)
+   url = 'http://common.farm1.smdg.ca/Forms/PlatformVideoFeed?platformUrl=http%3A//feed.theplatform.com/f/dtjsEC/9H6qyshBZU3E/categories%3Fpretty%3Dtrue%26byHasReleases%3Dtrue%26byCustomValue%3D%7Bplayertag%7D%7Bz/Showcase%20Video%20Centre%7D%26sort%3DfullTitle&callback='
+   html = self.getRequest(url)
+   a = json.loads(html[1:len(html)-1])['items']
+   pid = 'http://data.media.theplatform.com/media/data/Category/%s' % geurl
+   wewait = True
+   for b in a:
+      if b['parentId'] == pid:
+          name = b['title']
+          url  = b['id'].rsplit('/',1)[1]
+          ilist = self.addMenuItem(name,'GE', ilist, url, self.addonIcon, self.addonFanart, {}, isFolder=True)
+   return(ilist)
+
+
+  def getAddonEpisodes(self,url,ilist):
+    addonLanguage  = self.addon.getLocalizedString
+    self.defaultVidStream['width']  = 640
+    self.defaultVidStream['height'] = 360
+    geurl = uqp(url)
+    url = 'http://feed.theplatform.com/f/dtjsEC/9H6qyshBZU3E?count=true&byCategoryIDs=%s&startIndex=1&endIndex=100&sort=pubDate|desc&callback=' % geurl
+    html  = self.getRequest(url)
+    a = json.loads(html)['entries']
+    if len(a) == 0:
+     url = 'http://common.farm1.smdg.ca/Forms/PlatformVideoFeed?platformUrl=http%3A//feed.theplatform.com/f/dtjsEC/9H6qyshBZU3E/categories%3Fpretty%3Dtrue%26byHasReleases%3Dtrue%26byCustomValue%3D%7Bplayertag%7D%7Bz/Showcase%20Video%20Centre%7D%26sort%3DfullTitle&callback='
+     html = self.getRequest(url)
+     a = json.loads(html[1:len(html)-1])['items']
+     pid = 'http://data.media.theplatform.com/media/data/Category/%s' % geurl
+     wewait = True
+     for b in a:
+      if b['parentId'] == pid:
+        if wewait == True:
```

Comment:
I still think in C++, I'm not so young you know. In fact I'm so old, I do the compare for True because some early C compilers didn't handle that correctly - 0 = False, Anything else = True.